### DISCUSSION TITLE
Explanation of Settings

### DISCUSSION THREAD

**Discussion by Geniy83** [j-kaltes/Juggluco#97] [Q&A]
There are a number of questions:
1. Tell me in more detail what the Android flag does (in previous versions it was Android 13)? In what cases should it be installed?
2. What disables "Stream" in menu 3? Did I understand correctly what disables the raw data graph?
3. The "Stream history" flag allows the graph to be filled in when the Bluetooth connection was established

**Comment by j-kaltes** [comment]
1. The "_Android_" option does what the help says, it let Android make the connection with the sensor. When it is not set, Juggluco asks for a connection and every time the connection is disconnected, Juggluco asks again for it. When it is said, Juggluco asks once for the connection with a special flag to say that Android should connect automatically. Mostly the last method takes more time for a connection to establish, because Android does do it with less priority. But Android 13 contained a bug that made that a callback on disconnection was sometimes not called and because Juggluco asks again for a connection when that callback is called, Juggluco doesn't ask again for a connection when Android doesn't call that callback https://developer.android.com/reference/android/bluetooth/BluetoothGattCallback#onConnectionStateChange(android.bluetooth.BluetoothGatt,%20int,%20int). This had the effect that Juggluco didn't ask again for a connection so that no connection was established.
2. I don't understand your question. Unsetting "_Stream_" in the right middle menu, means that the stream values are not shown, as you can easily verify if you have a Libre 2 or 3 sensor.
3. Is that a question? Yes, if you turn off "Stream History" history values that otherwise only are received by scanning are also received by Bluetooth.

**Comment by Geniy83** [comment]
The Libre 3 sensor is now installed
![photo_2023-11-16_07-01-07](https://github.com/j-kaltes/Juggluco/assets/86565643/4ecb2698-01f2-473d-a80f-d45a6155a660)
First photo with stream enabled
![photo_2023-11-16_07-01-19](https://github.com/j-kaltes/Juggluco/assets/86565643/51f2a9bc-a5c9-4cf9-961b-4e97ad61ab29)
Second photo with the stream turned off
What data does each graph show?

**Comment by j-kaltes** [comment]
The first shows _Stream_ and _History_, the second only _History_. You can also turn off _History_ in the right middle menu.
Stream is a value for every minute, history (for Libre 3 sensors) every 5 minutes. The history values are always of the past, the current value is _Stream_.
As said in the help of left menu->Sensor, "_Stream History_" has no effect for Libre 3 sensors; history of Libre 3 sensors is always received via Bluetooth.

**Comment by Geniy83** [comment]
Thanks a lot

**Comment by j-kaltes** [comment]
In the next version Juggluco uses "_Bluetooth History_" instead of "_Stream History_". Maybe that is clearer.
